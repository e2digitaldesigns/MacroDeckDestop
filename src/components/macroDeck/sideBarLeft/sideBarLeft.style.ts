import styled from "styled-components";

export const SidebarLeft = styled.section`
  position: fixed;
  top: ${props => props.theme.modules.sidebarLeft.position.top};
  height: ${props => props.theme.modules.sidebarLeft.sizes.height};
  width: ${props => props.theme.modules.sidebarLeft.sizes.width};
  border-top: 0.0625rem solid
    ${props => props.theme.modules.sidebarLeft.colors.borderTop};
  border-right: 0.0625rem solid
    ${props => props.theme.modules.sidebarLeft.colors.borderRight};
  overflow: hidden;
  background-color: ${props => props.theme.modules.sidebarLeft.colors.bg};
  transition: width 1s;
  z-index: 999;
`;
