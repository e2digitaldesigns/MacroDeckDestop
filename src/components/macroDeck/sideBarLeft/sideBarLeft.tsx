import React from "react";
import Navigation from "./navigation/navigation";
import * as Styled from "../../../styles/app.style";

export const MacroDeckSidebarLeft: React.FC = () => {
  return (
    <Styled.Sidebarleft>
      <Navigation />
    </Styled.Sidebarleft>
  );
};
