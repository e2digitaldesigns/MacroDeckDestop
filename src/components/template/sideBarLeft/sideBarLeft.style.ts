import styled from "styled-components";

interface IntSidebarleft {
  showMenuLeft?: boolean;
}

export const Sidebarleft = styled.section<IntSidebarleft>`
  position: fixed;
  top: 80px;
  height: calc(100vh - 105px);
  width: ${props => props.theme.sizes.sidebarLeft.width};
  border-top: 0.0625rem solid ${props => props.theme.colors.border.default};
  border-right: 0.0625rem solid ${props => props.theme.colors.border.default};
  overflow: hidden;
  background-color: ${props => props.theme.colors.panel.bg};
  transition: width 1s;
  z-index: 999;
`;
