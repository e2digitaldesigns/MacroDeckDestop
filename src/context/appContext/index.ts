/* istanbul ignore file */
export * from "./appContext";
export * from "./defaultState";
