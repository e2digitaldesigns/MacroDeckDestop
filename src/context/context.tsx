import React from "react";
import { TemplateContextProvider } from "./templateContext";
import { ThemeContextProvider } from "./themeContext";

export interface IThemeComponents {
  TemplateProvider: React.ElementType;
  ThemeProvider: React.ElementType;
}

export const ApplicationContext: React.FC & IThemeComponents = () => <div />;

ApplicationContext.TemplateProvider = TemplateContextProvider;
ApplicationContext.ThemeProvider = ThemeContextProvider;
