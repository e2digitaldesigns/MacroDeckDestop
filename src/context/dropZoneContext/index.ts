export * from "./dropZoneContext";
export * from "./defaultState";
