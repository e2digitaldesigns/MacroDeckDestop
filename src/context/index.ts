export * from "./appContext";
export * from "./dropZoneContext";
export * from "./globalContext";
export * from "./templateContext";
export * from "./themeContext";
