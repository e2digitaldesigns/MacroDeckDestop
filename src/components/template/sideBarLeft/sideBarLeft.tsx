import React from "react";
import * as Styled from "./sideBarLeft.style";

export const TemplateSidebarLeft: React.FC = () => {
  return <Styled.Sidebarleft></Styled.Sidebarleft>;
};
