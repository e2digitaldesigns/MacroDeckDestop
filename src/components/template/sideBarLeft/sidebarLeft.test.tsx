import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { TemplateSidebarLeft } from "./sideBarLeft";

import { AppContext } from "../../../context/context";

const testSetup = () => {
  return render(
    <AppContext.ThemeProvider>
      <TemplateSidebarLeft />
    </AppContext.ThemeProvider>
  );
};

describe("<Template Sidebar Left/>", () => {
  const wrapper = testSetup();
  it("Should render without errors", () => {
    const component = wrapper;
    expect(component).toMatchSnapshot();
  });
});
