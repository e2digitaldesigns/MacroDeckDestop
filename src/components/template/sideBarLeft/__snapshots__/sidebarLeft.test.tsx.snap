// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Template Sidebar Left/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-papXJ fZXHHA"
      >
        <div
          class="sc-dkzDqf BNvcl"
        >
          <div
            class="sc-hKMtZM cGcfbh"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
              />
            </svg>
            <div>
              Viewing
              <span
                class="sc-eCYdqJ hiIcEh"
              >
                0
                / 
                0
                 
              </span>
            </div>
            <div
              class="sc-jSMfEi jIPZWC"
            >
              <span>
                Show All
              </span>
            </div>
          </div>
          <div>
            <input
              class="sc-gKXOVf bJCSrY"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="sc-bczRLJ eVealv"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-papXJ fZXHHA"
    >
      <div
        class="sc-dkzDqf BNvcl"
      >
        <div
          class="sc-hKMtZM cGcfbh"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
            />
          </svg>
          <div>
            Viewing
            <span
              class="sc-eCYdqJ hiIcEh"
            >
              0
              / 
              0
               
            </span>
          </div>
          <div
            class="sc-jSMfEi jIPZWC"
          >
            <span>
              Show All
            </span>
          </div>
        </div>
        <div>
          <input
            class="sc-gKXOVf bJCSrY"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="sc-bczRLJ eVealv"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
